src/savegame_json.cpp

```diff
@@ -2736,6 +2750,13 @@ void vehicle::deserialize( JsonIn &jsin )
     data.read( "airworthy", flyable );
     data.read( "summon_time_limit", summon_time_limit );
     data.read( "magic", magic );
+
+    smart_controller_cfg = cata::nullopt;
+    if( data.has_object( "smart_controller" ) ) {
+        smart_controller_cfg = smart_controller_config();
+        smart_controller_cfg -> deserialize( data.get_object( "smart_controller" ) );
+    }
```

Comment:
```suggestion
    smart_controller_cfg = cata::nullopt;
    data.read( "smart_controller", smart_controller_cfg );
```
`cata::optional` can be read directly when the value it contains supports the `deserialize` method.